### DISCUSSION TITLE
[Problem]ERROR DirectML initialization failed

### DISCUSSION THREAD

**Discussion by sugi0310** [vladmandic/sdnext#2711] [Q&A] [answered]
I am using Vladmandic in a Windows 10 environment.
When I completely cleared Vladmandic and reinstalled it, the following error occurred.

16:33:39-356005 ERROR    DirectML initialization failed: No module named 'torch_directml'
16:33:39-360993 ERROR    DirectML initialization failed: {e}

Even if I specify `--reinstall `in the startup options and install again, the problem is not resolved.

I'm having trouble finding a solution. Sorry if this is a basic problem. 
I have also attached the log for your reference.

[sdnext.log](https://github.com/vladmandic/automatic/files/13927304/sdnext.log)

**Comment by vladmandic** [comment, marked as answer]
your command line had BOTH `--use-rocm` and `--use-directml` - its one or the other and rocm flag had higher priority so torch_directml never got installed.

**Reply by sugi0310** [reply]
As you pointed out, I started it according to the method you suggested and the problem was resolved.

Thank you for your thorough explanation, even though it was a beginner's mistake.

**Comment by brknsoul** [comment]
As vlad said, you need to use only `--use-directml` in Windows with an AMD GPU.

To fix your problem, run `webui.bat --use-directml --reinstall`
Next time you run SD.Next, use `webui.bat --use-directml`
If you have an 8gb GPU, you might want to add `--medvram`.
Also, a couple of other suggested args would be `--debug` to get more helpful output in the console and `--autolaunch` to auto launch the browser.

`webui.bat --use-directml --medvram --debug --autolaunch`

**Reply by sugi0310** [reply]
Thank you for your reply. 
The problem was resolved successfully.
Thanks for the advice regarding boot options. 
I will try.
